### DISCUSSION TITLE
How do you supply variables via an extension?

### DISCUSSION THREAD

**Discussion by astroDimitrios** [quarto-dev/quarto-cli#10917] [Q&A] [answered]
### Description

I'm trying to create an extension which I'd like to have provide a set of translation variables for the [html templates](https://github.com/astroDimitrios/carpentries-quarto/blob/main/_extensions/carpentries/templates/title-block.html) (I am trying to convert our current mustache templates to be compatible with quarto using variables/metadata).

I can see docs on how to include [metadata](https://quarto.org/docs/extensions/metadata.html) in an extension but is there any way of including variables like you can with a `_variables.yml ` in a normal project?

**Comment by mcanouil** [comment, marked as answer]
Short answer (as far as I know) is no.

You might want to look at the following extension which is a bit related to what I think you are trying to do.
https://github.com/ArthurData/quarto-i18n

I'm not sure to understand why you want to use `_variables.yml` instead of actual metadata here.

**Reply by astroDimitrios** [reply]
Many thanks - that extension is very cool and I will have to try it out. I guess I was just trying to avoid putting a lot of metadata in the extension yaml file but if that's the only way then so be it :)

**Reply by mcanouil** [reply]
`metadata-files` should work for the end user which would allow to split the metadata into files.
